src/ast/mod.rs

```diff
@@ -7766,6 +7766,27 @@ impl fmt::Display for RenameTable {
     }
 }
 
+/// table object for insertion
+#[derive(Debug, Clone, PartialEq, PartialOrd, Eq, Ord, Hash)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+#[cfg_attr(feature = "visitor", derive(Visit, VisitMut))]
+pub enum TableObject {
+    // for simple table name
+    TableName(#[cfg_attr(feature = "visitor", visit(with = "visit_relation"))] ObjectName),
+
+    // for Clickhouse [table functions](https://clickhouse.com/docs/en/sql-reference/table-functions)
```

Comment:
```suggestion
    /// Table specified as a function.
    /// Example:
    /// ```sql
    /// INSERT INTO TABLE FUNCTION remote('localhost', default.simple_table)
    /// ```
    /// [Clickhouse](https://clickhouse.com/docs/en/sql-reference/table-functions)
```